lib/connection.ts

```diff
@@ -121,26 +125,26 @@ export class Connection {
     return new Promise((resolve, reject) => {
       if (!this.isOpen()) {
 
-        let onOpen: Func<rhea.EventContext, void>;
-        let onClose: Func<rhea.EventContext, void>;
+        let onOpen: Func<EventContext, void>;
+        let onClose: Func<EventContext, void>;
         let waitTimer: any;
 
         const removeListeners: Function = () => {
           clearTimeout(waitTimer);
-          this._connection.removeListener(ConnectionEvents.connectionOpen, onOpen);
```

Comment:
for your internal listening i'd still listen to the underlying connection directly. this prevents you from getting messed up by a user calling `removeAllListeners()` or something like that.